### ISSUE TITLE
There shouldn't be gain and loss at the same time in func report

### ISSUE THREAD

**Issue by zgfzgf** [yearn/yearn-vaults#368]
https://github.com/yearn/yearn-vaults/blob/8978c89095aa3a05224103b067b95b8c1627de5f/contracts/Vault.vy#L1633

if loss > 0 && gain >0
I feel a little unreasonable. 
There shouldn't be gain and loss at the same time

In this way, it can be better
if loss == 0 || gain ==0

**Comment by steffenix** [comment]
We keep it possible for extreme accounting cases when we have to "hack" the accounting, the strategist is the one responsible for using the function properly.

**Comment by zgfzgf** [comment]
I want to rethink today.
**Vault should controller strategist.**

for example.
Token X put in pool. harvest  get  Token Y. Token Y transfer to Token X.

when gain = 50 Token X (harvest get X). loss = 100 Token X (liquidatePosition) but  strategy get 50 * performanceFee. It is unreasonable. strategy  should get 0 Token.

when gain = 50 Token X. loss = 20 Token X. strategy get (50-20) * performanceFee. It is more reasonable.

so:
if loss > 0 && gain >0:
- loss -= min(loss,  gain);
- gain -=min(loss,  gain);

**Comment by steffenix** [comment]
That's not for usual scenarios but if we need to hack things around in order to save some locked funds because of a bug for example.
